server/datastore/mysql/sessions.go

```diff
@@ -16,6 +17,9 @@ func (d *Datastore) SessionByKey(ctx context.Context, key string) (*fleet.Sessio
 	session := &fleet.Session{}
 	err := sqlx.GetContext(ctx, d.reader, session, sqlStatement, key)
 	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, ctxerr.Wrap(ctx, notFound("Session").WithName("<key redacted>"))
```

Comment:
Do you mean to add `.WithName("<key redacted>")` to prevent future devs to add the session to the error?

Reply:
The error gets logged, so the idea is that this still returns a `notFound` error that correctly represents what happened, but using a redacted key `WithName("<key redacted>")` so that it won't log the key when logging the error.

Reply:
Sorry, am still confused (didn't have my coffee today)

`return nil, ctxerr.Wrap(ctx, notFound("Session").WithName("<key redacted>"))`

There's no `err` there.

Reply:
This `notFound("Session").WithName("<key redacted>")` is an error, it represents a row that was expected to be there but wasn't found (i.e. we transform the `sql.ErrNoRows` error to that more general error).

Reply:
Yes, I think I understand what you mean, that we could've just not call `.WithName(...)`? I guess so, although it still indicates what type of loading was attempted (i.e. via the key instead of id).

Reply:
+1, was just checking in case I missed some other package or process setting the `Name`.